### ISSUE TITLE
Batch norm decay may be too low with sequential patches

### ISSUE THREAD

**Issue by mikowals** [beopst/dcn.tf#3]
When training full DCN a moving average decay of 0.9 might be too low for 8 patches run sequentially through the fine layers.  Each call of fine_layers() will run an update of the moving averages so you are effectively using a decay of 0.43 ( 0.9 ^ 8 patches) per 64 image training batch.  So the moving averages in the fine layers for validation are probably very skewed by the training batch that last ran.

I think this is another reason to concat all the patches into one batch for fine layers rather than call in series.  

```
fine_features = fine_layers(tf.concat( 0, k_patches))
fine_features = tf.split( 0, N_PATCHES, fine_features)
```

**Comment by beopst** [comment]
Thank you for the good point! I revised the code to concat all the patches. 
To expect shuffling effects on batch normalization,
- add sorted=False argument to tf.nn.top_k 
- shuffle top_k_idxs along the dimension of patches

```
top_k_values, top_k_idxs = tf.nn.top_k(ops.flatten(M), N_PATCHES, sorted=False)
top_k_idxs = tf.random_shuffle(tf.transpose(top_k_idxs))
top_k_idxs = tf.transpose(top_k_idxs)
```

I expect that DCN gives slightly better performance than before. I will update the results soon.
